PowerEditor/src/TinyXml/tinyxml.cpp

```diff
@@ -551,20 +551,37 @@ void TiXmlElement::SetAttribute( const TCHAR * name, const TCHAR * _value )
 	}
 }
 
-void TiXmlElement::Print( FILE* cfile, int depth ) const
+const char one_ws[] = " ";
+const char four_ws[] = "    ";
+const char eol[] = "\r\n";
+
+const generic_string tagOpenSymb = TEXT("<");
+const generic_string tagOpenSlash = TEXT("</");
+const generic_string tagCloseSymb = TEXT(">");
+
+const char tagCloseChar[] = ">";
+const char tagCloseSymbFinal[] = " />";
+
+void TiXmlElement::Print( CFile& cfile, int depth ) const
 {
 	int i;
 	for ( i=0; i<depth; i++ )
 	{
-		generic_fprintf( cfile, TEXT("    ") );
+		cfile.Write(four_ws, 4);
```

Comment:
Instead of using the magic number "4" you can use `std::size(four_ws)`.
Also, you can add an overloaded method to CFile that accepts only a string and does `strlen` on it to determine the size.

By the way, I'm not sure using a named constant improves the readability of code here.
Here we have simple and short string literals that are easy to understand.
Also,  reducing such modifications could help merging with a newer version of TinyXML library.

Reply:
TinyXML libary in Notepad++ hasn't been updated since long time ago, and won't be updated in the future. There's no bug in it from the usage of Notepad++, so why bother?

cfile.Write needs a pointer of char and `cfile.Write("    ", 4);` won't work, so a named variable cannot be avoid. Since this 4 ws value is used in the different place, I prefer to declare it as constant outside of this function. Use `4` instead of std::size(four_ws) is to prevent from using cast. They are not magic number for these trivial `const char*`.

Reply:
You can add an overload `bool Write(const char* wbuf, size_t buf_size);`

Reply:
I see @mere-human has proposed the same thing as me here: https://github.com/donho/notepad-plus-plus/commit/a8ec068965b2a0eed010b53d88a18d14bb678e58#commitcomment-57345690 - to make an overloaded Write method of the CFile class that accepts directly a `std::string` - will make things much clearer.

Reply:
Anyhow there is no possibility to update, see https://sourceforge.net/projects/tinyxml, apart from changing to https://github.com/leethomason/tinyxml2, which uses UTF-8 and probably needs to be adapted for windows utf-16.